### PR TITLE
feat(APP-3630): Update radio card and checkbox card to accept children property and render when selected/checked

### PR DESCRIPTION
**Pull request by shan8851**
## Description

Update radio card and checkbox card to accept children property and render when selected/checked

Task: [APP-3630](https://aragonassociation.atlassian.net/browse/APP-3630)

## Type of change

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

-   [x] I have selected the correct base branch.
-   [x] I have performed a self-review of my own code.
-   [x] I have commented my code, particularly in hard-to-understand areas.
-   [x] I have made corresponding changes to the documentation.
-   [x] My changes generate no new warnings.
-   [ ] Any dependent changes have been merged and published in downstream modules.
-   [x] I ran all tests with success and extended them if necessary.
-   [x] I have updated the `CHANGELOG.md` file in the root folder of the package after the [UPCOMING] title and before
        the latest version.
-   [ ] I have tested my code on the test network.

[APP-3630]: https://aragonassociation.atlassian.net/browse/APP-3630?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,8 +7,15 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 
 ## [Unreleased]
 
+### Added
+
+-   Added optional `children` property to `RadioCard` and `CheckboxCard` core components. When selected children will be
+    rendered
```
Comment (cgero-eth) [inline CHANGELOG.md:13]:
```suggestion
-   Add optional `children` property to `RadioCard` and `CheckboxCard` core components displayed only when component is selected
```

File: src/core/components/forms/checkboxCard/checkboxCard.test.tsx [hunk 2]
```diff
@@ -62,4 +62,10 @@ describe('<CheckboxCard /> component', () => {
         render(createTestComponent({ checked }));
         expect(screen.getByTestId(IconType.CHECKBOX_INDETERMINATE)).toBeVisible();
     });
+
+    it('renders the children when checkbox is checked', () => {
```
Comment (cgero-eth) [inline src/core/components/forms/checkboxCard/checkboxCard.test.tsx:66]: Can we also have a test to verify that the `children` property is instead hidden when the component is not checked?

File: src/core/components/forms/checkboxCard/checkboxCard.tsx [hunk 3]
```diff
@@ -40,10 +40,26 @@ export interface ICheckboxCardProps extends ComponentProps<'button'> {
      * Id of the checkbox.
      */
     id?: string;
+    /**
+     * Additional children to render when the checkbox is checked.
+     */
+    children?: ReactNode;
```
Comment (cgero-eth) [inline src/core/components/forms/checkboxCard/checkboxCard.tsx:46]: Can we also make the `description` property optional for the `CheckboxCard` component?

File: src/core/components/forms/radioCard/radioCard.tsx [hunk 4]
```diff
@@ -67,18 +71,26 @@ export const RadioCard = forwardRef<HTMLButtonElement, IRadioCardProps>((props,
...
+                        {description && <p className={baseTextClasses}>{description}</p>}
                     </div>
                     {tag && <Tag {...tag} />}
                 </div>
-                <span className="h-full">
+                <span
+                    className={classNames({
+                        'h-full': description,
+                    })}
+                >
```
Comment (cgero-eth) [inline src/core/components/forms/radioCard/radioCard.tsx:93]:
```suggestion
                <span className={classNames({ 'h-full': description })}>
```

File: src/core/components/forms/radioCard/radioCard.test.tsx [hunk 5]
```diff
@@ -72,4 +72,21 @@ describe('<RadioCard/> component', () => {
...
         expect(screen.getByRole('radio')).toHaveValue(value);
     });
+
+    it('renders children when radio button is checked', async () => {
+        const user = userEvent.setup();
+        render(
+            createTestComponent({
+                children: <div data-testid="child-content">child content</div>,
+            }),
+        );
```
Comment (cgero-eth) [inline src/core/components/forms/radioCard/radioCard.test.tsx:82]: We can just pass a text as children and avoid multiple line-breaks
```suggestion
        const user = userEvent.setup();
        
        const children = 'test-children';
        render(createTestComponent({ children }));
```

File: src/core/components/forms/checkboxGroup/checkboxGroup.stories.tsx [hunk 6]
```diff
@@ -57,4 +58,38 @@ export const CheckboxCardVariant: Story = {
...
+ */
+export const CheckboxCardVariantWithChildren: Story = {
+    render: (props) => (
+        <CheckboxGroup {...props}>
+            <CheckboxCard
+                label="First label"
+                description="First description"
+                avatar="first"
+                tag={{ label: 'First', variant: 'info' }}
+            >
```
Comment (cgero-eth) [inline src/core/components/forms/checkboxGroup/checkboxGroup.stories.tsx:72]: I'd avoid having multiple stories for the same feature, I'd only keep this on the `RadioCard` / `CheckboxCard` component is enough

File: src/core/components/forms/radioCard/radioCard.stories.tsx [hunk 7]
```diff
@@ -22,6 +22,41 @@ type Story = StoryObj<typeof RadioCard>;
...
+        label: 'Option one',
+        description: 'The best option ever',
+        tag: { label: 'Platinum' },
+    },
+};
+
+/**
+ * Usage of the `RadioCard` component with children when selected
+ */
+export const WithChildrenWhenCSelected: Story = {
```
Comment (cgero-eth) [inline src/core/components/forms/radioCard/radioCard.stories.tsx:59]:
```suggestion
export const WithChildrenWhenSelected: Story = {
```

**Review by cgero-eth — COMMENTED** [review]
Thanks for the changes @shan8851!

File: src/core/components/forms/checkboxCard/checkboxCard.test.tsx (hunk 2 above)
Comment (shan8851) [reply src/core/components/forms/checkboxCard/checkboxCard.test.tsx:66]: So it seems jest is not playing nice with tailwind and doesn't recognise the hidden and block classes, so for now I have left this test out

**Review by cgero-eth — APPROVED** [review]

**Pull request merged**